enterprise/server/remote_execution/dirtools/dirtools_test.go

```diff
@@ -10,18 +11,322 @@ import (
+				BytesTransferred: 1,
+			},
+		},
+		{
+			name: "FileAndDirectory",
+			cmd: &repb.Command{
+				OutputFiles:       []string{"a/fileA.txt"},
+				OutputDirectories: []string{"a"},
+			},
+			directoryPaths: []string{
+				"a",
+			},
+			fileContents: map[string]string{
+				"a/fileA.txt": "a",
+			},
+			symlinkPaths: map[string]string{},
+			expectedResult: &repb.ActionResult{
+				OutputDirectories: []*repb.OutputDirectory{
+					{
+						Path: "a",
+						TreeDigest: &repb.Digest{
+							SizeBytes: 85,
+							Hash:      "895545df6841b7efb2e9cc903a4eac7a60c645199be059f6056817ae6feb071d",
+						},
+					},
+				},
+				OutputFiles: []*repb.OutputFile{
+					{
+						Path: "a/fileA.txt",
+						Digest: &repb.Digest{
+							SizeBytes: 1,
+							Hash:      "ca978112ca1bbdcafac231b39a23dc4da786eff8147c4e72b9807785afee48bb",
+						},
+					},
+				},
+			},
+			expectedInfo: &dirtools.TransferInfo{
+				FileCount:        2,
+				BytesTransferred: 84,
+			},
+		},
+		{
+			name: "SymlinkToFile",
+			cmd: &repb.Command{
+				OutputFiles: []string{
+					"fileA.txt",
+					"linkA.txt",
+				},
+			},
+			directoryPaths: []string{},
+			fileContents: map[string]string{
+				"fileA.txt": "a",
+			},
+			symlinkPaths: map[string]string{
+				"linkA.txt": "fileA.txt",
+			},
+			expectedResult: &repb.ActionResult{
+				OutputFiles: []*repb.OutputFile{
+					{
+						Path: "fileA.txt",
+						Digest: &repb.Digest{
+							SizeBytes: 1,
+							Hash:      "ca978112ca1bbdcafac231b39a23dc4da786eff8147c4e72b9807785afee48bb",
+						},
+					},
+				},
+				OutputFileSymlinks: []*repb.OutputSymlink{
+					{
+						Path:   "linkA.txt",
+						Target: "fileA.txt",
+					},
+				},
+			},
+			expectedInfo: &dirtools.TransferInfo{
+				FileCount:        1,
+				BytesTransferred: 1,
+			},
+		},
+		{
+			name: "SymlinkToDirectory",
+			cmd: &repb.Command{
+				OutputFiles: []string{"a/fileA.txt"},
+				OutputDirectories: []string{
+					"a",
+					"linkA",
+				},
+			},
+			directoryPaths: []string{
+				"a",
+			},
+			fileContents: map[string]string{
+				"a/fileA.txt": "a",
+			},
+			symlinkPaths: map[string]string{
+				"linkA": "a",
+			},
+			expectedResult: &repb.ActionResult{
+				OutputDirectories: []*repb.OutputDirectory{
+					{
+						Path: "a",
+						TreeDigest: &repb.Digest{
+							SizeBytes: 85,
+							Hash:      "895545df6841b7efb2e9cc903a4eac7a60c645199be059f6056817ae6feb071d",
+						},
+					},
+				},
+				OutputFiles: []*repb.OutputFile{
+					{
+						Path: "a/fileA.txt",
+						Digest: &repb.Digest{
+							SizeBytes: 1,
+							Hash:      "ca978112ca1bbdcafac231b39a23dc4da786eff8147c4e72b9807785afee48bb",
+						},
+					},
+				},
+				OutputDirectorySymlinks: []*repb.OutputSymlink{
+					{
+						Path:   "linkA",
+						Target: "a",
+					},
+				},
+			},
+			expectedInfo: &dirtools.TransferInfo{
+				FileCount:        2,
+				BytesTransferred: 84,
+			},
+		},
+		{
+			name: "DanglingSymlinkIsIgnored",
```

Comment:
Added comment.

I think unresolved symlink is only supported as outputs and not as inputs. See https://cs.opensource.google/bazel/bazel/+/master:src/test/java/com/google/devtools/build/lib/remote/RemoteExecutionServiceTest.java;l=216-231?q=RemoteExecution&ss=bazel

Also, the flag is enabled by default since Bazel 6.0.0 so it's not an opt-in feature.